src/client/client.ts

```diff
@@ -85,7 +92,7 @@ function createDispatchers(
 ) {
   return innerActionNames.reduce((dispatchers, name) => {
     dispatchers[name] = function (...args: any[]) {
-      store.dispatch(
+      return store.dispatch(
```

Comment:
Until we decide how to surface errors, how about we leave dispatchers unchanged as void functions?

```suggestion
      store.dispatch(
```

In my RFC I wondered about returning promises, but perhaps as a first step we should focus on the existing subscription/update system rather than introducing this new path for state updates.

This could be something like adding an error parameter to subscription callbacks, so where currently you do

```js
client.subscribe((state: State | null) => {});
```

we could implement errors so you do

```js
client.subscribe((state: State | null, error?: ActionError) => {});
```

Existing consumers would be unimpacted, but users could upgrade to get that error object.

Reply:
Seems good to me. Only tactical concern is that, without making some decisions in this PR on how to expose the error parameters to the subscription callbacks, testability becomes a bit tricky.

Reply:
I think testing it only where currently relevant is OK:

1. Test the reducer in isolation.
2. Test the store with the transient handling middleware in isolation.
3. Test the client, but only to make sure transients don’t end up in state, something like

    ```js
    client.moves.invalid();
    expect(client.getState().transients).toBeUndefined();
    ```

4. Test the master, but for now only to ensure transients don’t get sent in the update payload.

Once the master is emitting an error event and the client is doing whatever it ends up doing, we can then test those.

Reply:
Sweet. Set the ActionResult type alias back to `void`, reverted the added `return`s, and modified the client test to only check that no transients remained (with actionable TODOs to add the appropriate assertions in the subsequent PRs for #723).